lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -722,6 +733,20 @@ instance Arbitrary (PostTransactionFeeData t) where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
+instance Arbitrary PostExternalTransactionData where
+    arbitrary = genericArbitrary
```

Comment:
I think we should probably avoid `genericArbitrary` if we can. It can make it hard to reason about what values are generated.

In this case, we should be able to inline your definition from `Arbitrary ByteString` below.

Reply:
indeed. Inlined `Arbitrary ByteString` inside `PostExternalTransactionData` instance